spec/unit/actions/collect_data_product_spec.rb

```diff
@@ -5,23 +5,12 @@
 # LICENSE file in the root directory of this source tree.
 
 describe GoodData::LCM2::CollectDataProduct do
-  it 'Has GoodData::Bricks::DefaultDataProductMiddleware class' do
-    GoodData::LCM2::CollectDataProduct.should_not be(nil)
-  end
-
   let(:client) { double(:client) }
   let(:domain) { double(:domain) }
   let(:gdc_logger) { double(:gdc_logger) }
   let(:data_product) { GoodData::DataProduct.new({}) }
 
-  before do
-    allow(client).to receive(:domain) { domain }
-    allow(gdc_logger).to receive(:info)
-    allow(domain).to receive(:data_products) { data_product }
-  end
-
-  context 'when data_product parameter is passed' do
-    let(:data_product_id) { 'data-product' }
+  shared_context 'data_product config' do
```

Comment:
There is no need for the new `shared_context`. The let can be simply defined inside the `describe`.